docs/decisions/0015-expiration-improvements.rst

```diff
@@ -0,0 +1,94 @@
+0015 Expiration and Cancellation Data Improvements
+**************************************************
+
+Status
+======
+Proposed - December 2023
+
+Context
+=======
+There are subtleties around the Assignment cancellation and expiration user
+experience in the enterprise learner-portal frontend that require a more nuanced
+and descriptive approach to the data we persist around these states, and
+how that data is represented in API response payloads for consumption
+by the learner-portal frontend.
```

Comment:
nit: might mention the repo name `frontend-app-learner-portal-enterprise` whenever "learner-portal" is referred (e.g., someone might get confused with `frontend-app-learner-portal-programs`?)